src/Nethermind/Nethermind.State/IVisitingWorldState.cs

```diff
@@ -0,0 +1,30 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Core.Crypto;
+using Nethermind.Evm.State;
+using Nethermind.Trie;
+
+namespace Nethermind.State;
+
+public interface IVisitingWorldState : IWorldState
```

Comment:
Do we want this to be under same interface hierarchy or use interface segregation?

Reply:
I tried to make separate interface for that initially. But in some places we need both IWorldState and Accept function. I thought splitting them will add too much complexity with questionable upsides

Reply:
This seems like something that can be moved to `IStateReader`.